### ISSUE TITLE
HUnit Exception lazy message escapes timeout

### ISSUE THREAD

**Issue by fendor** [feuerbach/tasty#280]
Example:

```haskell
import Test.Tasty
import Test.Tasty.HUnit

main :: IO ()
main = defaultMain spec

spec :: TestTree
spec =
  testGroup
    "Example"
    [ testCase "Equal lists" $ 
       badList @?= [3, 2, 1]
    ]

badList :: [Int]
badList = [1,2,3, endlessRecursion]

endlessRecursion :: a
endlessRecursion = endlessRecursion
```

This has interesting behaviour.
Trying to run this via:
```
> :main -t 1
```
will not terminate. It get's stuck on the testcase. The problem is the HUnit exception, which knows that the test-case failed (since the first element is already wrong), but tries to show the `Expected` and `Actual` values. However, the `actual` value can't be shown, as the last element is essentially bottom.

```haskell
timeout 10000 (badList @?= [3,2,1])
> *** Exception: HUnitFailure (Just (SrcLoc {srcLocPackage = "interactive", srcLocModule = "Ghci1", srcLocFile = "<interactive>", srcLocStartLine = 7, srcLocStartCol = 15, srcLocEndLine = 7, srcLocEndCol = 67})) "expected: [3,2,1]\n but got: [1,2,3,
``` 

I think we can mitigate the issue by forcing the string value before throwing the exception.

**Comment by feuerbach** [comment]
Thanks for reporting and diagnosing this. However, your fix in #281 was specific for HUnit, whereas this is a general issue that's best addressed in tasty proper.

**Comment by fendor** [comment]
You are absolutely right, thank you!
